httpx/_transports/default.py

```diff
@@ -28,6 +28,7 @@
 from types import TracebackType
 
 import httpcore
+from httpcore.backends.base import SOCKET_OPTION
```

Comment:
Let's not import that type definition, it's not documented public API.

It's be okay to instead define the socket option types here.

```python
SOCKET_OPTION = typing.Union[
    typing.Tuple[int, int, int],
    typing.Tuple[int, int, typing.Union[bytes, bytearray]],
    typing.Tuple[int, int, None, int],
]
```